AMM/SPECIFICATION.md

```diff
@@ -135,41 +134,65 @@ This sub-section details what a user is able to do, e.g., click a button and "x,
     8. Requires a deadline (block height limit)
 
-#### Previews
+### Previews
 
-##### `preview_add_liquidity()`
+#### `preview_add_liquidity()`
 
 1. Returns the amount of the other asset to input and the liquidity asset amount to receive after an add liquidity operation
     1. If the asset pair of the pool is set 
     2. Requires the amount of an asset to input to [`add_liquidity`](#add_liquidity) 
         > **NOTE** If any liquidity in the contract already exists, than the amount of the other asset is calculated based on the ratio of the assets. Otherwise, the ratio is assumed to be 1.   
 
-##### `preview_swap_exact_input()`
+#### `preview_swap_exact_input()`
 
 1. Returns the minimum output amount to receive after a [`swap_exact_input`](#swap_exact_input) and whether the output asset reserves are sufficient for the swap
     1. If the asset pair of the pool is set 
     2. Requires an exact amount of either asset to sell
 
-##### `preview_swap_exact_output()`
+#### `preview_swap_exact_output()`
 
 1. Returns the maximum input amount for a [`swap_exact_output`](#swap_exact_output) and whether the input asset reserves are sufficient for the swap
     1. If the asset pair of the pool is set
     2. If the output asset reserves are sufficient for the swap
     3. Requires an exact amount of either asset to buy
 
-#### State Checks
+### State Checks
 
-##### `balance()`
+#### `balance()`
 
 1. Returns the asset balance of the sender in the contract 
     1. If the asset pair of the pool is set
     2. Requires the asset identifier to return the balance for
 
-##### `pool_info()`
+#### `pool_info()`
 
 1. Returns the pool info, i.e., the identifiers and amounts of assets and the liquidity pool asset amount 
     1. If the asset pair of the pool is set
 
-## Sequence Diagram
+## Scripts
+
+### `atomic-add-liquidity`
+
+1. Deposits pool assets and adds liquidity
+    1. If desired liquidity is more than 0
+    2. If [`deposit`](#deposit) and [`add_liquidity`](#add_liquidity) conditions are met
+
+### `swap-exact-input`
+
+1. Swaps assets along a route by specifying exact input for each swap
+    1. If the route has at least 2 assets
+    2. If the exact input amount is specified for each swap
+    3. If the AMM has a pool for each subsequent asset pair in route
+    4. If [`swap_exact_input`](#swap_exact_input) conditions are met
+
+### `swap-exact-output`
+
+1. Swaps assets along a route by specifying exact output for each swap
+    1. If the route has at least 2 assets
+    2. If the exact output amount is specified for each swap
+    3. If the AMM has a pool for each subsequent asset pair in route
+    4. If [`swap_exact_output`](#swap_exact_output) conditions are met
+
+# Sequence Diagram
 
 ![AMM Sequence Diagram](.docs/amm-sequence-diagram.png)
```

Comment:
Shouldn't the diagram be altered to show that you can construct the exchange and then deposit, the deposit is used to create the pool and then you can keep depositing / do other things. 
The way this reads to me is that you can construct and somehow during the construction you add a pool, after you add a pool you can do things that do not link to that pool (no lines) and hence there is the assumption of knowledge that you can do those things.